From f8fb087b33a82590689dc914d0711ad856bc744c Mon Sep 17 00:00:00 2001
From: Bin Han <bin.b.han@intel.com>
Date: Wed, 14 Jan 2015 21:10:32 +0800
Subject: [PATCH 239/255] atomisp2: enable CVF API is changed

See ICG gerrit https://icggerrit.ir.intel.com/#/c/33558/,
the CVF control interface is changed and this patch is to switch to
the new API for atomisp2 driver.

Change-Id: I8e4e7cba3b7bd1dd25ced031de9fc24dfd906f3d
Tracked-On: #H1503907016
Signed-off-by: Bin Han <bin.b.han@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/36221
Reviewed-by: Peng Qiao <peng.qiao@intel.com>
Reviewed-by: Tian Shu Qiu <tian.shu.qiu@intel.com>
Reviewed-by: GuoqingX Zhang <guoqingx.zhang@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
---
 .../media/pci/atomisp2/atomisp_driver/atomisp_cmd.c       |  3 +--
 .../media/pci/atomisp2/atomisp_driver/atomisp_compat.h    |  3 ++-
 .../pci/atomisp2/atomisp_driver/atomisp_compat_css20.c    | 15 +++++++++++++--
 3 files changed, 16 insertions(+), 5 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index 3cf3c64..3b1000f 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -4297,8 +4297,7 @@ static int __enable_continuous_mode(struct atomisp_sub_device *asd,
 	atomisp_css_preview_enable_online(asd, ATOMISP_INPUT_STREAM_GENERAL,
 			!enable);
 	atomisp_css_enable_continuous(asd, enable);
-	atomisp_css_enable_cont_capt(enable,
-				!asd->continuous_viewfinder->val);
+	atomisp_css_enable_cvf(asd, asd->continuous_viewfinder->val);
 
 	if (atomisp_css_continuous_set_num_raw_frames(asd,
 			asd->continuous_raw_buffer_size->val)) {
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat.h
index e537bd1..68cfcff 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat.h
@@ -333,7 +333,8 @@ void atomisp_css_video_enable_online(struct atomisp_sub_device *asd,
 void atomisp_css_enable_continuous(struct atomisp_sub_device *asd,
 							bool enable);
 
-void atomisp_css_enable_cont_capt(bool enable, bool stop_copy_preview);
+void atomisp_css_enable_cvf(struct atomisp_sub_device *asd,
+							bool enable);
 
 int atomisp_css_input_configure_port(struct atomisp_sub_device *asd,
 				mipi_port_ID_t port,
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
index 14bc008..94fb9cf 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
@@ -498,6 +498,8 @@ static void __dump_stream_config(struct atomisp_sub_device *asd,
 	dev_dbg(isp->dev, "stream_config.online=%d.\n", s_config->online);
 	dev_dbg(isp->dev, "stream_config.continuous=%d.\n",
 			s_config->continuous);
+	dev_dbg(isp->dev, "stream_config.disable_cont_viewfinder=%d.\n",
+			s_config->disable_cont_viewfinder);
 	dev_dbg(isp->dev, "stream_config.channel_id=%d.\n",
 			s_config->channel_id);
 	dev_dbg(isp->dev, "stream_config.init_num_cont_raw_buf=%d.\n",
@@ -2228,9 +2230,18 @@ void atomisp_css_enable_continuous(struct atomisp_sub_device *asd,
 	}
 }
 
-void atomisp_css_enable_cont_capt(bool enable, bool stop_copy_preview)
+void atomisp_css_enable_cvf(struct atomisp_sub_device *asd,
+				bool enable)
 {
-	sh_css_enable_cont_capt(enable, stop_copy_preview);
+	struct atomisp_stream_env *stream_env =
+		&asd->stream_env[ATOMISP_INPUT_STREAM_GENERAL];
+	int i;
+
+	if (stream_env->stream_config.disable_cont_viewfinder != !enable) {
+		stream_env->stream_config.disable_cont_viewfinder = !enable;
+		for (i = 0; i < IA_CSS_PIPE_ID_NUM; i++)
+			stream_env->update_pipe[i] = true;
+	}
 }
 
 int atomisp_css_input_configure_port(
-- 
1.9.1

